File: Example/ICACrashReporter/GoogleAnalyticsObject.m
//
//  GoogleAnalyticsObject.m
//  ICACrashReporter
//
//  Created by Jelle Vandebeeck on 03/12/15.
//  Copyright © 2015 Jelle Vandebeeck. All rights reserved.
//

#import <ICACrashReporter/GoogleAnalytics.h>

#import "GoogleAnalyticsObject.h"

@implementation GoogleAnalyticsObject

- (instancetype)init
{
    self = [super init];
    if (self) {
        [self setupCrashReporter];
    }
    return self;
}

#pragma mark - Setup

- (void)setupCrashReporter {
    // Setup the reporters.
    NSArray *reporters = @[
                           [ICAConsoleCrashReporter new],
                           [ICAGoogleAnalyticsCrashReporter new],
                           ];
    ICAMultiCrashReporter *multiReporter = [[ICAMultiCrashReporter alloc] initWithReporters:reporters];
    [ICACrashReporter initAndStartWithInstance:multiReporter];
    
    // Run some reporting tools.
    [ICACrashReporter logBreadcrumb:@"test %@ %f %@", @"x", 23.3, @"TEST"];
    [ICACrashReporter logServiceFailure:404 serviceUrl:@"http://icapps.com" httpMethod:@"GET"];
    [ICACrashReporter setUserIdentifier:@"user A"];
    [ICACrashReporter logEvent:@"new event"];
    [ICACrashReporter logExtraData:@"key" value:@"value"];
    [ICACrashReporter logException:[NSException exceptionWithName:@"new exception" reason:@"crashed" userInfo:nil]];
    
    // Try the transaction controller.
    ICACrashReporterTransactionController *controller = [ICACrashReporter transactionController];
    [controller startTransaction];
    [controller stopTransaction];
    [controller cancelTransaction];
}

@end

File: Example/Swift/ICACrashReporter/GoogleAnalyticsObject.swift
//
//  GoogleAnalyticsObject.swift
//  ICACrashReporter Example
//
//  Created by Jelle Vandebeeck on 03/12/15.
//  Copyright © 2015 Jelle Vandebeeck. All rights reserved.
//

import Foundation
import ICACrashReporter

class GoogleAnalyticsObject {
    
    // MARK: - Init
    
    init() {
        setupCrashReporter()
    }
    
    // MARK: - Setup
    
    private func setupCrashReporter() {
        // Setup the reporters.
        let reporters = [
            ICAConsoleCrashReporter(),
            ICAGoogleAnalyticsCrashReporter()
        ]
        let provider = ICAMultiCrashReporter(reporters: reporters)
        ICACrashReporter.sharedInstance().provider = provider
        
        // Run some reporting tools.
        ICACrashReporter.sharedInstance().logServiceFailureWithStatusCode(404, serviceURL: "http://icapps.com", HTTPMethod: "GET")
        ICACrashReporter.sharedInstance().logBreadcrumb("KAWA BUNGA")
        ICACrashReporter.sharedInstance().userIdentifier = "123"
        ICACrashReporter.sharedInstance().logEvent("BOOM!")
        ICACrashReporter.sharedInstance().logKey("a", value: "b")
        
        // Log an exception.
        let exception = NSException(name: "BOOM", reason: "Inconsistent design", userInfo: nil)
        ICACrashReporter.sharedInstance().logException(exception)
        
        // Use the transaction controller.
        let controller = ICACrashReporter.sharedInstance().instantiateTransactionController()
        controller.start()
        controller.stop()
        controller.cancel()
    }
}